src/main/java/org/folio/edge/controller/exception/ExceptionHandlerController.java

```diff
@@ -77,4 +83,25 @@ public Error handleException(Exception e) {
     return new ResponseEntity<>(new Error().error(INVALID_REQUEST_ERROR_TYPE), HttpStatus.valueOf(e.getStatus()));
   }
 
+  @ExceptionHandler(FeignException.class)
+  public ResponseEntity<InnReachResponseDTO> handleFeignStatusException(FeignException feignException) {
+    var status = HttpStatus.resolve(feignException.status());
+    String body = feignException.contentUTF8();
+    InnReachResponseDTO innReachResponseDTO = null;
+    try {
+      innReachResponseDTO = objectMapper.readValue(body, InnReachResponseDTO.class);
+    } catch (JsonProcessingException e) {
+      log.error("Unexpected exception: {}" + e.getMessage());
+      innReachResponseDTO = failed(body);
+      return new ResponseEntity<>(innReachResponseDTO, status);
```

Comment:
return is redundant in this block, it's just a duplicate of method's return

Reply:
Fixed